plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -761,12 +763,85 @@ public void testBigQueryExternalTable()
+            assertThat(query("SELECT * FROM test." + biglakeExternalTable)).matches("SELECT * FROM tpch.region");
+
+            assertUpdate("DROP TABLE test." + biglakeExternalTable);
+            assertQueryReturnsEmptyResult("SELECT * FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + biglakeExternalTable + "'");
+
+            // BigLake tables should not run queries, since they are read directly using the storage read API.
+            verifyThatStreamApiWasUsed(biglakeExternalTable);
+        }
+        finally {
+            onBigQuery("DROP EXTERNAL TABLE IF EXISTS test." + biglakeExternalTable);
+        }
+    }
+
+    @Test
+    public void testExternalObjectTable()
+    {
+        String objectTableConnectionId = bigQueryConnectionId;
+
+        String objectExternalTable = "test_object_external" + randomNameSuffix();
+
+        try {
+            onBigQuery("CREATE EXTERNAL TABLE test." + objectExternalTable +
+                       " WITH CONNECTION `" + objectTableConnectionId + "`" +
+                       " OPTIONS (object_metadata = 'SIMPLE', uris = ['gs://" + gcpStorageBucket + "/tpch/tiny/region.csv'])");
+            assertQuery("SELECT table_type FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + objectExternalTable + "'", "VALUES 'BASE TABLE'");
+
+            assertThat(query("SELECT uri FROM test." + objectExternalTable)).succeeds();
+
+            assertUpdate("DROP TABLE test." + objectExternalTable);
+            assertQueryReturnsEmptyResult("SELECT * FROM information_schema.tables WHERE table_schema = 'test' AND table_name = '" + objectExternalTable + "'");
+
+            // BigLake tables should not run queries, since they are read directly using the storage read API.
+            verifyThatStreamApiWasNotUsed(objectExternalTable);
+        }
+        finally {
+            onBigQuery("DROP EXTERNAL TABLE IF EXISTS test." + objectExternalTable);
+        }
+    }
+
+    private void verifyThatStreamApiWasUsed(String externalTable)
+    {
+        verifyHowManyTimesTableWasReferencedInJobs(externalTable, 0);
+    }
+
+    private void verifyThatStreamApiWasNotUsed(String externalTable)
+    {
+        verifyHowManyTimesTableWasReferencedInJobs(externalTable, 1);
+    }
+
+    private void verifyHowManyTimesTableWasReferencedInJobs(String externalTable, long noOfReferences)
+    {
+        assertThat(bigQuerySqlExecutor.executeQuery("""
+                 SELECT count(*) FROM region-us.INFORMATION_SCHEMA.JOBS WHERE EXISTS(
+                     SELECT * FROM UNNEST(referenced_tables) AS referenced_table
+                         WHERE referenced_table.table_id = '%s')
+                """.formatted(externalTable)).getValues())
+                .extracting(fieldValues -> fieldValues.getFirst().getLongValue())
+                .containsExactly(noOfReferences);
+    }
+
```

Comment:
We could change this helper method to return count something like:
```java
    private long getTableReferenceCountInJob(String tableId)
    {
        return bigQuerySqlExecutor.executeQuery("""
                 SELECT count(*) FROM region-us.INFORMATION_SCHEMA.JOBS WHERE EXISTS(
                     SELECT * FROM UNNEST(referenced_tables) AS referenced_table
                         WHERE referenced_table.table_id = '%s')
                """.formatted(tableId)).streamValues()
                .map(fieldValues -> fieldValues.getFirst().getLongValue())
                .collect(onlyElement());
    }
```
We don't need to add other 2 helper methods in my opinion.

Reply:
done